HIP/hip-1056.md

```diff
@@ -0,0 +1,1769 @@
+3. As a verifier for a network, I want a single, self-contained stream of data with aggregated signature signed by a
+   threshold of stake weights, so that I can cost-effectively confirm that the stream represents the consensus output of
+   the network.
+4. As a downstream developer, I want a low latency feed that provides the state changes associated with transaction(s)
+   within a block, so that I can interact with state changes with the smallest possible delay.
+
+## Specification
+
+The Block Stream will be defined by the continuous transmission of Block information defined as follows.
+- Each `Block` represents all inputs and outputs of the consensus network for zero or more whole `Rounds`. The maximum
+  number of rounds per block will be configurable and start at 1 round per block. It could be increased later if
+  necessary for performance reasons or if round become very frequent and small.
+    - A block *may* be zero rounds because the last block prior to a network "freeze" must communicate the state hash
+      (via a block proof) of the state immediately prior to network shutdown. This is only possible with an extra empty
+      block because the state hash in each block proof is the state hash at the *beginning* of that block/round. This
+      delay of state hash communication is required to simultaneously meet Hedera performance goals and also meet EVM
+      expectations for immediate access to recent (the last 256) block hash values.
+- A `Round` is a sequence of zero or more `Events` that have reached consensus at the same time. Round length in real
+  world time depends on internet latency and event creation rate. In Hedera mainnet today it is around 1 round per
+  second. In the future this could be shorter to lower latency, but would never make sense to be less than ping time.
+- An `Event` is a sequence of zero or more `EventTransactions` submitted by a single node.
+- Each `EventTransaction` is either a `SignedTransaction` or a system transaction.
+- A `SignedTransaction` is transmitted as a byte array, contains the signed transaction bytes of a user-submitted
+  transaction exactly as processed by the network, and is followed by transaction outputs and state changes resulting
+  from that transaction.
+- System transactions are represented by explicitly defined protobuf messages and used for consensus node to consensus
+  node communication.
+
+These concepts are core to the Hashgraph consensus mechanism and have direct impacts on the structure of data in the 
+streams.
+
+Block streams are built out of Items. Those items can be delivered in a continuous stream containing one or more blocks.
+Or they can be delivered as a single block using the `Block` message type container. If a single block is delivered as
+a file it expected to be either a raw protobuf `Block` message with file name of `<BLOCK_NUMBER>.blk` e.g.
+`0000000000000000001.blk` . With total of 19 digits to allow for 2^63 blocks. Or a compressed file e.g.
```

Comment:
Currently Block file names are 36-digit strings padding with leading zeroes. [FileBlockItemWriter](https://github.com/hashgraph/hedera-services/blob/develop/hedera-node/hedera-app/src/main/java/com/hedera/node/app/blocks/impl/FileBlockItemWriter.java#L222)

Reply:
Removed this section as it was confused for API and is more an internal block node consideration

Reply:
Just for completion.  I have no idea why the consensus node chose 36-digits, but we chose to restrict file names in the block node to what a Java `long` can represent for efficiency.  It's easy enough to trim or pad the extra 17 `0`s, if needed.